src/cljs/harja/views/urakka/suunnittelu/tehtavat.cljs

```diff
@@ -23,110 +28,330 @@
-           {:otsikko "Sovittu koko urakka jäljellä" :nimi :sovittuja-jaljella :tyyppi :string :muokattava? (constantly false) :leveys 3})
-       (when (or (not t/sopimuksen-tehtavamaarat-kaytossa?) sopimukset-syotetty?) 
-         {:otsikko [:<> 
-                    [:div "Suunniteltu määrä"] 
-                    [:div "hoitokausi"]] :nimi :maara :tyyppi :numero :muokattava? kun-yksikko :leveys 3})
-       {:otsikko "Yksikkö" :nimi :yksikko :tyyppi :string :muokattava? (constantly false) :leveys 2}]
-      (:atomi atomi)])])
+  [rivi]
+  (if-not (boolean? rivi)
+    (let [{:keys [yksikko]} rivi] 
+      (not (or 
+             (nil? yksikko)
+             (= "" yksikko)
+             (= "-" yksikko))))
+    rivi))
+
+(defn vertaile-onko-samat 
+  [vertailumaara [_ maara]] 
+  (= vertailumaara maara))
+
+(defn- samat-sopimusmaarat? 
+  [rivi]
+  (let [vertailumaara (-> rivi :sopimuksen-tehtavamaarat first second)]
+    (every? (r/partial vertaile-onko-samat vertailumaara) (:sopimuksen-tehtavamaarat rivi))))
+
+(defn- kun-kaikki-samat
+  [rivi]
+  (if (:joka-vuosi-erikseen? rivi)
+    false
+    rivi))
+
+(defn tallenna!
+  [e! sopimukset-syotetty? alueet-vai-maarat rivi]
+  (let [tuck-event (cond
+                     (and (= :alueet alueet-vai-maarat)
+                       sopimukset-syotetty?)
+                     t/->TallennaMuuttunutAluemaara
+                     
+                     (and (= :maarat alueet-vai-maarat)
+                       sopimukset-syotetty?)
+                     t/->TallennaTehtavamaara
+
+                     (and (= :alueet alueet-vai-maarat)
+                       (not sopimukset-syotetty?))
+                     t/->TallennaSopimuksenAluemaara
+                     
+                     (and (= :maarat alueet-vai-maarat)
+                       (not sopimukset-syotetty?))
+                     t/->TallennaSopimuksenTehtavamaara)] 
+    (e! (tuck-event rivi))))
+
+(defn vali->viiva [nimi] (str/replace (str/lower-case nimi) " " "-"))
+
+(defn vetolaatikko-komponentti-
+  [_ _ {:keys [vanhempi id yksikko nimi] :as _rivi}]
+  (let [joka-vuosi-erikseen? (r/cursor t/taulukko-tila [:maarat vanhempi id :joka-vuosi-erikseen?])]
+    (fn [e! app {:keys [vanhempi id] :as rivi}]
+      [:div.tehtavat-maarat-vetolaatikko 
+       [:div.vetolaatikko-ruksi 
+        [kentat/tee-kentta {:tyyppi :checkbox 
+                            :teksti "Haluan syöttää joka vuoden erikseen"
+                            :valitse!
+                            (fn [v]
+                              (let [ruksittu? (.. v -target -checked)]
+                                (swap! t/taulukko-tila assoc-in [:maarat vanhempi id :joka-vuosi-erikseen?] ruksittu?)
+                                (e! (t/->JokaVuosiErikseenKlikattu ruksittu? vanhempi id))))}
+         joka-vuosi-erikseen?]]
+       [:div.flex-row.tasaa-alas.loppuun
+        (doall 
+          (for [vuosi (range
```

Comment:
Range antaa vectorin. Eli vuosi-> vuodet :)

Reply:
tässähän tuo for iteroi tuon rangen läpi vuosi kerrallaan, joka bindautuu tuohon vuosi-symboliin